docs/debugging-dotnet-lambda-functions.md

```diff
@@ -0,0 +1,153 @@
+
+## Install and configure prerequisites
+
+1. Install the [AWS Toolkit for Visual Studio Code](https://github.com/aws/aws-toolkit-vscode#getting-started).
+2. Install the [C# extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp). This extension gives VS Code the ability to debug .NET Core applications.
+3. Launch Visual Studio Code and open a SAM application or create a new one. <!-- TODO: Link to separate doc with instructions. -->
+
+    Note: Open the folder that contains `template.yaml`.
+
+4. Open a terminal in the folder containing `template.yaml` and set up the debugger by running the following commands:
+
+    * replace `<CODE_URI>` with the *absolute* path matching the `CodeUri` property from `template.yaml` for the resource that you wish to debug.
+    * Replace `dotnetcore2.1` with the framework identifier for the runtime that you are targetting.
+
+    ```bash
+    mkdir <CODE_URI>/.vsdbg
+    docker run --rm --mount type=bind,src=<CODE_URI>/.vsdbg,dst=/vsdbg --entrypoint bash lambci/lambda:dotnetcore2.1 -c "curl -sSL https://aka.ms/getvsdbgsh | bash /dev/stdin -v latest -l /vsdbg"
+    ```
+
+## Configure your debugger
+
+1. Open `<workspace folder root>/.vscode/launch.json` (create a new file if it does not already exist), and add the following contents:
+
+    * If desired, replace `5679` with the port that you wish to use for debugging.
+
+    ```jsonc
+    {
+        "version": "0.2.0",
+        "configurations": [
+            {
+                "name": "SamLocalDebug",
+                "type": "coreclr",
+                "request": "attach",
+                "processId": "1",
+                "pipeTransport": {
+                    "pipeProgram": "sh",
+                    "pipeArgs": [
+                        "-c",
+                        "docker exec -i $(docker ps -q -f publish=5679) ${debuggerCommand}"
+                    ],
+                    "debuggerPath": "/tmp/lambci_debug_files/vsdbg",
+                    "pipeCwd": "<CODE_URI>"
+                },
+                "windows": {
+                    "pipeTransport": {
+                        "pipeProgram": "powershell",
+                        "pipeArgs": [
+                            "-c",
+                            "docker exec -i $(docker ps -q -f publish=5679) ${debuggerCommand}"
+                        ],
+                        "debuggerPath": "/tmp/lambci_debug_files/vsdbg",
+                        "pipeCwd": "<CODE_URI>"
+                    }
+                },
+                "sourceFileMap": {
+                    "/var/task": "<CODE_URI>"
+                }
+            }
+        ]
+    }
+    ```
+
+2. Launch Visual Studio Code and open the folder containing your application.
+3. Press `Ctrl+Shift+D` or click the `Debug` icon to open the debug viewlet:
+
+    ![Debug Icon](./images/view_debug.png)
+
+4. Select `.NET Core: Remote Attach` from the drop-down menu at the top of the viewlet.
```

Comment:
1. The entry was renamed to `SamLocalDebug`
1. Consider explicitly writing something along the lines of "Don't press the Play button yet. The steps below will build and run the program that this section will attach to."